adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -363,55 +364,84 @@ protected void handleRemoteReceiverOpen(final ProtonReceiver receiver, final Pro
+     *  <li> The tenant that the device belongs to is disabled for the AMQP adapter. </li>
+     * </ul>
+     * 
      * @param sender The link to use for sending commands to devices.
-     * @param connection The connection instance with the device.
+     * @param authenticatedDevice The authenticated device or {@code null} if the device is not authenticated.
      */
-    protected void handleRemoteSenderOpenForCommands(final ProtonSender sender, final ProtonConnection connection) {
+    protected void handleRemoteSenderOpenForCommands(final ProtonSender sender, final Device authenticatedDevice) {
         if (sender.getRemoteSource() == null || Strings.isNullOrEmpty(sender.getRemoteSource().getAddress())) {
             // source address is required
-            closeLinkWithError(sender, new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Missing source address"));
+            closeLinkWithError(sender,
+                    new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Missing source address"));
         } else {
-            final ResourceIdentifier address = ResourceIdentifier.fromString(sender.getRemoteSource().getAddress());
-            if (!CommandConstants.isCommandEndpoint(address.getEndpoint())) {
-                closeLinkWithError(sender, new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Only the command endpoint is supported"));
-            } else {
-                final Device authenticatedDevice = connection.attachments().get(AmqpAdapterConstants.KEY_CLIENT_DEVICE, Device.class);
+            try {
+                final ResourceIdentifier address = ResourceIdentifier.fromString(sender.getRemoteSource().getAddress());
                 validateAddress(address, authenticatedDevice).compose(validAddress -> {
-                    return createCommandConsumer(sender, validAddress, authenticatedDevice).map(consumer -> {
-
-                        final String tenantId = validAddress.getTenantId();
-                        final String deviceId = validAddress.getResourceId();
-                        sender.setSource(sender.getRemoteSource());
-
-                        sender.setQoS(sender.getRemoteQoS());
-                        HonoProtonHelper.setCloseHandler(sender, remoteDetach -> {
-                            sendDisconnectedTtdEvent(tenantId, deviceId, authenticatedDevice);
-                            closeCommandConsumer(tenantId, deviceId);
-                            onLinkDetach(sender);
+                    if (!CommandConstants.isCommandEndpoint(validAddress.getEndpoint())) {
+                        closeLinkWithError(sender, new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                                "Only the command endpoint is supported"));
+                    } else {
+                        final Future<TenantObject> tenantTracker = getTenantConfiguration(validAddress.getTenantId(), null);
```

Comment:
I do not think that this is the right place to do this. Why would we allow a device to connect to the AMQP adapter in the first place, if the AMQP adapter has been disabled for the device's tenant?

Reply:
yes that make sense. will update the patch